go/tools/gazelle/rules/generator.go

```diff
@@ -35,28 +36,70 @@ type Generator interface {
 }
 
 // NewGenerator returns an implementation of Generator.
-func NewGenerator() Generator {
-	return new(generator)
+//
+// "goPrefix" is the go_prefix corresponding to the repository root.
+// See also https://github.com/bazelbuild/rules_go#go_prefix.
+func NewGenerator(goPrefix string) Generator {
+	return &generator{
+		// TODO(yugui): Support another resolver to cover the pattern 2 in
+		// https://github.com/bazelbuild/rules_go/issues/16#issuecomment-216010843
+		r: structuredResolver{goPrefix: goPrefix},
+	}
 }
 
-type generator struct{}
+type generator struct {
+	r labelResolver
+}
 
 func (g *generator) Generate(rel string, pkg *build.Package) ([]*bzl.Rule, error) {
+	r, err := g.generate(rel, pkg)
+	if err != nil {
+		return nil, err
+	}
+	return []*bzl.Rule{r}, nil
+}
+
+func (g *generator) generate(rel string, pkg *build.Package) (*bzl.Rule, error) {
 	kind := "go_library"
 	name := "go_default_library"
```

Comment:
I might pull the "go_default_library" up to a constant with a comment about this expectation.